SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Storage/Storage.cs

```diff
@@ -235,7 +207,7 @@ internal Storage(IStorageAdapter storageAdapter)
             }
```

Comment:
Two questions:
(1) If we are throwing the same StorageException(errorMessage, e) for all four types of exceptions, can we simplify this code to check for exception type and throw the exception once?
(2) What's the reason ignoring exception outside the four types here? Should we catch and re-throw?